data/scripting/infrastructure.lua

```diff
@@ -281,8 +286,9 @@ function check_trees_rocks_poor_hamlet(player, sf, warehouse, waretable_rocks, w
       player:send_to_inbox(_("No rocks nearby"), _("There are no rocks near to your starting position. Therefore, you receive extra resources for bootstrapping your economy."))
    end
    -- adding exactly one forester
-   if not has_trees then
+   if n_trees < min_trees then
+      waretable_trees.log = math.max(0, waretable_trees.log - n_trees) -- reduce additional logs if there are trees
```

Comment:
... but here `waretable_trees.log`

I try to use `a_table[variable]` or `a_table[3]` but `a_table.label` (not `a_table["label"]`), but I mixed here :no_mouth:

Reply:
I prefer using the [] always as basically there is no difference in a key to the table whether it being a number, a letter or a string like in this case. Lua allows all of them in the same table afaik.

Reply:
You are also free to write `f["immovable"]:has_attribute("tree")`, for lua a `wl.map.Field` is just a special table. But I guess we agree this would not look nice.

Reply:
correct I realized that myself just seconds before your post.    
So I only miss how to credit your work in our credit list?

Reply:
only nickname "aDiscoverer" is fine